kubevirtmng/ui/DiskList.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import $ from "jquery"
+
+
+class Volume extends React.Component {
+  constructor() {
+      super();
+
+      this.state = {
+         disks: []
+      }
+
+      this.fetchData = this.fetchData.bind(this);
+    };
+
+   componentWillMount() {
+      this.fetchData();
+   };
+
+   fetchData() {
+        $.ajax({
+            type: "GET",
+            dataType: 'json',
+            url: 'http://localhost:8083/v1/volumes/'+this.props.data.name+"/",
+            success: function(response){
+            console.log(response);
+            this.setState({disks: response.disks})
```

Comment:
.bind(this) gets very trying if you end up with more levels of nesting, which is common, and may happen as your app gets larger.

let self = this

Is a common metapattern (before the functions), then you can use "self.setState({ ...") from any level of anonymous nesting